### ISSUE TITLE
React `Authenticator.SetupTOTP` crashes on form fields lookup

### ISSUE THREAD

**Issue by calebpollman** [aws-amplify/amplify-ui#1785]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

NA

### Please describe your bug.

The `Authenticator` crashes when a custom `formFields` object is passed to the `Authenticator` if it does not contain values for `SetupTOTP.QR`:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'totpIssuer')
    at y (SetupTOTP.js:1:1)
    at renderWithHooks (react-dom.development.js:16141:1)
    at mountIndeterminateComponent (react-dom.development.js:20838:1)
    at beginWork (react-dom.development.js:22342:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:4157:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4206:1)
    at invokeGuardedCallback (react-dom.development.js:4270:1)
    at beginWork$1 (react-dom.development.js:27243:1)
    at performUnitOfWork (react-dom.development.js:26392:1)
    at workLoopSync (react-dom.development.js:26303:1)
y @ SetupTOTP.js:1
renderWithHooks @ react-dom.development.js:16141
mountIndeterminateComponent @ react-dom.development.js:20838
beginWork @ react-dom.development.js:22342
callCallback @ react-dom.development.js:4157
invokeGuardedCallbackDev @ react-dom.development.js:4206
invokeGuardedCallback @ react-dom.development.js:4270
beginWork$1 @ react-dom.development.js:27243
performUnitOfWork @ react-dom.development.js:26392
workLoopSync @ react-dom.development.js:26303
renderRootSync @ react-dom.development.js:26271
performSyncWorkOnRoot @ react-dom.development.js:25924
flushSyncCallbacks @ react-dom.development.js:11982
(anonymous) @ react-dom.development.js:25490
```

There is a workaround, specify custom values for the lookup to resolve to:
```
const formFields = {
// ...other form field key/values
  setupTOTP: {
    QR: {
      totpIssuer: <CUSTOM_ISSUER>,
      totpUsername: <CUSTOM_USERNAME>,
    },
  },
}

const App = () => (
  <Authenticator formFields={formFields}>
    {/* your app */}
  </Authenticator>
)
```

### What's the expected behaviour?

`Authenticator` should not crash.

### Help us reproduce the bug!

Pass an empty object as the value of the `formFields` prop:
```
<Authenticator formFields={{}}>
  <App />
</Authenticator>
```

### Code Snippet

```typescript
// Put your code below this line.

```

**Comment by slaymance** [comment]
This issue is fixed in the latest [@aws-amplify/ui-react@2.15.7](https://github.com/aws-amplify/amplify-ui/releases/tag/%40aws-amplify%2Fui-react%402.15.7) release.

**Comment by nk9** [comment]
I was running into this `totpIssuer` problem and updating `ui-react` fixed it for me. Thanks!
